lib/Model/Binary.cpp

```diff
@@ -173,15 +173,27 @@ MetaAddressRangeSet Binary::executableRanges() const {
   return ExecutableRanges;
 }
 
-Identifier Function::name() const {
-  using llvm::Twine;
+std::optional<Identifier>
+Function::tryGetName(const TypeOfGlobalNameCache &Names,
+                     const NamingConfiguration &Configuration) const {
   if (not CustomName().empty()) {
+    revng_assert(Names.contains(CustomName()), "Cache is outdated?");
     return CustomName();
+
   } else {
-    auto AutomaticName = (Twine("_function_") + toIdentifier(Entry())).str();
+    std::string AutomaticName = Configuration.unnamedFunctionPrefix()
+                                + toIdentifier(Entry());
+    if (Names.contains(AutomaticName)) {
+      // COLLISION!!! EVERYONE, PANIC!
+      // jokes aside, I'm not sure what to do here.
```

Comment:
You need to keep adding `_`.
No point in having `tryGetName`, let's just have `name` which adds as many `_` as necessary.

It seems to me that you're trying to avoid handling this case, but this is a perfectly normal situation that we need to handle silently.

Reply:
Done.